src/pages/home/settings/Settings.tsx

```diff
@@ -32,22 +34,47 @@ const Settings: FunctionComponent = () => {
     confirmPassword: "",
   };
 
-  const validationSchema = Yup.object({
-    currentPassword: Yup.string().required("Current password is required"),
-    newPassword: commonYupValidation.newPassword.when("currentPassword", {
-      is: (currentValue: string) => currentValue,
-      then: Yup.string()
-        .required("New password is required")
-        .notOneOf(
-          [Yup.ref("currentPassword")],
-          "New password cannot be the same as the current password"
-        ),
-    }),
-    confirmPassword: commonYupValidation.confirmPassword.when("newPassword", {
+  /**
+   * If there is a current password, the new password is required and it can't be the same as the current password.
+   */
+  const newPassword = commonYupValidation.newPassword.when("currentPassword", {
+    is: (currentValue: string) => currentValue,
+    then: Yup.string()
+      .required("New password is required")
+      .notOneOf(
+        [Yup.ref("currentPassword")],
+        "New password cannot be the same as the current password"
+      ),
+  });
+
+  /**
+   * If there is a new password, the confirmation password is required and it has to match the new password.
+   */
+  const confirmPassword = commonYupValidation.confirmPassword.when(
+    "newPassword",
+    {
       is: (currentValue: string) => currentValue,
       then: Yup.string().required("Must match new password"),
-    }),
-  });
+    }
+  );
+
+  /**
+   * Defines the validation schema based on the user's login method.
+   * If the user is logged in using social login, only the new password and the confirmation password are required.
+   * If the user logged in using email and password, the current password is also required.
```

Comment:
Should we provide some type of copy for the social login situation? Seems strange to have the password fields without any other indication that it will set up their email login... This doesn't need email verification as well?

Reply:
I think this is one of those things where we can get feedback on that and add if needed. You are correct, this doesn't need email verification. It was discussed that for MVP we allow them to do this and provide a different approach post-MVP.